src/app/components/AddSource.svelte

```diff
@@ -0,0 +1,189 @@
+<script lang="ts">
+  import worker from "../spawn-worker";
+  import Button from "./Button.svelte";
+  import SplitButton from "./SplitButton.svelte";
+  import ManualEntry from "./ManualEntry.svelte";
+  const UPLOAD_INPUT_ID = "upload-input";
+
+  const leftText = "UPLOAD";
+  const rightText = "DIRECT ENTRY";
```

Comment:
I have nuanced comments about this. Constants are good for software maintainability, but why are they good? Constants define one single place for some content to be defined. This is useful when:

 1.  the same value gets used in multiple different places (don't repeat yourself, easier to change the value throughout the entire codebase if needed, since you only have to change it in one place); or
 2. the constant, such as a integer, declares a _non-obvious_ value — then you can give it a meaningful name (e.g., `WEEKS_IN_A_YEAR` is more easy to understand than `52` in a codebase)!
 3. The use of the constant in one context is related to its use in another context. It helps signify to the programmer which values are related.

See: `UPLOAD_INPUT_ID` — it has a non obvious value that may need to be changed in the future, and is used in at least two places in the source code that have a related purpose.

These does not fit either of the above criteria.

The names `leftText` and `rightText`, in my opinion, _obscure_ the meaning of "upload" and "direct entry". And the value is a user-readable label that makes sense more sense in its true context.

As such, delete these here...

```suggestion
```

...then see a following comment.